core/src/main/java/com/netflix/conductor/core/execution/tasks/Join.java

```diff
@@ -56,15 +56,19 @@ public boolean execute(
                 break;
             }
             TaskModel.Status taskStatus = forkedTask.getStatus();
-            hasFailures = !taskStatus.isSuccessful() && !forkedTask.getWorkflowTask().isOptional();
+            hasFailures =
```

Comment:
the code at line 52 should already fetch the join task that is retried, so it will be in progress and will wait for it to complete.